pubsub/mqtte4k/mqtte4k.go

```diff
@@ -42,23 +42,37 @@ const (
 // mqttPubSub type allows sending and receiving data to/from MQTT broker.
 type mqttPubSub struct {
 	client   *mqtt.Client
+	auth     *mqtt.ClientConfig
 	metadata *metadata
 	logger   logger.Logger
 	topics   map[string]byte
 
-	ctx     context.Context
-	cancel  context.CancelFunc
-	backOff backoff.BackOff
-	svid    *jwtsvid.SVID
+	ctx      context.Context
+	cancel   context.CancelFunc
+	backOff  backoff.BackOff
+	svid     *jwtsvid.SVID
+	satToken string
 }
 
 // NewMQTTPubSub returns a new mqttPubSub instance.
 func NewMQTTE4KPubSub(logger logger.Logger) pubsub.PubSub {
 	return &mqttPubSub{
-		logger:          logger,
+		logger: logger,
 	}
 }
 
+func populateSATPassword(m *mqttPubSub) {
+	token, err := os.ReadFile(m.metadata.satTokenPath)
+	if err != nil {
+		panic(err)
```

Comment:
maybe add error log here saying, if file is not found. Have you added annotations?